### PR TITLE
OCPQE-13810 support Azure stack in gather-azure-resource step

### PR DESCRIPTION
**Pull request by MayXuQQ**
[OCPQE-13810](https://issues.redhat.com/browse/OCPQE-13810) suppor azure stack

### CODE REVIEW

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
@jinyunma  https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/36458/rehearse-36458-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive/1626411940857253888/artifacts/azure-stack-ipi-p3-f28-destructive/gather-azure-resource/build-log.txt 
thanks

**Comment by jinyunma** [comment]
Error in step `gather-azure-resource` build-log
```Running Command: az network lb address-pool address list --lb-name ci-op-y4qqvifk-12545-g8h9g --pool-name ci-op-y4qqvifk-12545-g8h9g --resource-group ci-op-y4qqvifk-12545 -o table
WARNING: Command group 'network lb address-pool address' is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: 'BackendAddressPool' object has no attribute 'load_balancer_backend_addresses'
```

I checked cluster on azure StackCloud, there is no field "loadBalancerBackendAddresses" when running command `az network lb address-pool show --lb-name logci8773-h77c7 --name logci8773-h77c7 --resource-group logci8773-h77c7-rg`, only have field "backendIpConfigurations"
```
# az network lb address-pool show --lb-name logci8773-h77c7 --name logci8773-h77c7 --resource-group logci8773-h77c7-rg | jq -r .backendIpConfigurations[].id
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-master-0-nic/ipConfigurations/pipConfig
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-master-2-nic/ipConfigurations/pipConfig
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-master-1-nic/ipConfigurations/pipConfig
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-worker-1-nic/ipConfigurations/pipConfig
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-worker-0-nic/ipConfigurations/pipConfig
/subscriptions/de7e09c3-b59a-4c7d-9c77-439c11b92879/resourceGroups/logci8773-h77c7-rg/providers/Microsoft.Network/networkInterfaces/logci8773-h77c7-worker-2-nic/ipConfigurations/pipConfig
```

In azure PublicCould, both loadBalancerBackendAddresses and backendIpConfigurations are available.
```
[fedora@preserve-jima custom-role]$ az network lb address-pool show --lb-name jimaaz411-d79m6 --name  jimaaz411-d79m6 --resource-group  jimaaz411-d79m6-rg | jq -r .loadBalancerBackendAddresses[].id
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-worker-westus-588x6-nicpipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-worker-westus-79tfd-nicpipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-worker-westus-ljwlw-nicpipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-master-0-nicpipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-master-2-nicpipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/loadBalancers/jimaaz411-d79m6/backendAddressPools/jimaaz411-d79m6/loadBalancerBackendAddresses/jimaaz411-d79m6-rg_jimaaz411-d79m6-master-1-nicpipConfig
[fedora@preserve-jima custom-role]$ az network lb address-pool show --lb-name jimaaz411-d79m6 --name  jimaaz411-d79m6 --resource-group  jimaaz411-d79m6-rg | jq -r .backendIpConfigurations[].id
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-worker-westus-588x6-nic/ipConfigurations/pipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-worker-westus-79tfd-nic/ipConfigurations/pipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-worker-westus-ljwlw-nic/ipConfigurations/pipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-master-0-nic/ipConfigurations/pipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-master-2-nic/ipConfigurations/pipConfig
/subscriptions/53b8f551-f0fc-4bea-8cba-6d1fefd54c8a/resourceGroups/jimaaz411-d79m6-rg/providers/Microsoft.Network/networkInterfaces/jimaaz411-d79m6-master-1-nic/ipConfigurations/pipConfig
```

Might need to adjust the command to get backendpool addresses on azure stack cloud to avoid the error?

**Comment by MayXuQQ** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive

**Comment by MayXuQQ** [comment]
@jinyunma https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/36458/rehearse-36458-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-azure-stack-ipi-p3-f28-destructive/1629752928543707136/artifacts/azure-stack-ipi-p3-f28-destructive/gather-azure-resource/build-log.txt

**Comment by jinyunma** [comment]
@MayXuQQ maybe we can run `az network lb address-pool show --lb-name $lb --name $lb --resource-group $RESOURCE_GROUP | jq -r .backendIpConfigurations[].id` instead of skipping it? Else need to check in lbs.json from Artifacts. WDYT?

**Comment by MayXuQQ** [comment]
/pj-rehearse  periodic-ci-openshift-verification-tests-master-installer-rehearse-4.13-installer-rehearse-azure

File: ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh [hunk 1]
```diff
@@ -26,12 +23,28 @@ function cli_Login() {
...
+            -n ${cloud_name} \
+            --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+            --suffix-storage-endpoint "${SUFFIX_ENDPOINT}" 
+        az cloud set -n ${cloud_name}
+        az cloud update --profile 2019-03-01-hybrid
     else
         echo "Unexpected cluster: ${CLUSTER_TYPE}"
         exit 1
     fi
+    AZURE_AUTH_CLIENT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientId)"
```
Comment (jinyunma) [inline ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh:45]: Based on [comments](https://github.com/openshift/release/pull/36488#discussion_r1118306689), we don't need to consider dev PPE env, so no need to move those 3 lines here.

Comment (gpei) [reply ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh:45]: I guess we can move them back as  AZURE_AUTH_* parameters are all set in the beginnning.

File: ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh [hunk 2]
```diff
@@ -26,12 +25,26 @@ function cli_Login() {
     elif [[ "${CLUSTER_TYPE}" == "azure4" ]]; then
         run_command "az cloud set --name AzureCloud" || return 1
     elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
-        echo "Unsupported cluster: ${CLUSTER_TYPE}"
-        exit 1
+        AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"
```
Comment (jianlinliu) [inline ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh:28]: I think we do not need this line here.

**Pull request merged**